### PR TITLE
Default network and storage configuration

### PR DESCRIPTION
**Pull request by MarceloSalazar**
Mbed OS 5.10 includes default network interfaces and storage block device drivers for platforms.

This PR introduces changes to:
- Update test configuration to use the default configuration
- Documentation & rebranding updates

Note, continues to be based on (MCC) Pelion Client library v1.4.0

This PR precedes the SMCC application here:

cc @dhwalters423 @bridadan @janjongboom

Please review asap - we aim to merge this today

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -68,65 +70,35 @@ Simple Mbed Cloud Client provides Greentea tests to test your Simple Mbed Cloud
...
-}
+"target.network-default-interface-type": "ETHERNET",
 ```
-For more examples and a template `mbed_app.json`, see [simple-mbed-cloud-client-template-restricted](https://github.com/ARMmbed/simple-mbed-cloud-client-template-restricted).
 
-4. You may need to delete your `main.cpp`.
+1. If your platform does not have a default storage component in `targets.json`, you can define this in `mbed_app.json`. Aditionally, you may want to double-check the storage pinout configuration for your platform in the `mbed_lib.json` file in `mbed-os/components/storage/blockdevice/`.
+   
+
+```
```
Comment (bridadan) [inline README.md]: I believe the backticks should be removed here, they are messing up the formatting for the rest of the doc.

Comment (MarceloSalazar) [reply README.md]: Good catch! Resolved now.

File: README.md [hunk 2]
```diff
@@ -68,65 +70,33 @@ Simple Mbed Cloud Client provides Greentea tests to test your Simple Mbed Cloud
 | `simple-connect` | Tests that the device successfully registers to Mbed Cloud using the specified storage, SOTP, and connectivity configuration. Tests that SOTP and the RoT is preserved over a reset and the device connects with a consistent device ID.  |
 
 ### Requirements
- Simple Mbed Cloud Client tests rely on the Python SDK to test the end to end solution. 
+ Simple Pelion Client tests rely on the Python SDK to test the end to end solution.
  To install the Python SDK:
 `pip install mbed-cloud-sdk`
  **Note:** The Python SDK requires Python 2.7.10+ / Python 3.4.3+, built with SSL support.
- 
+
```
[the remark this answers is not in the document]
Comment (MarceloSalazar) [reply README.md]: I've added a first step in instructions to ensure the developer starts from the application.

File: README.md [hunk 3]
```diff
@@ -68,65 +70,33 @@ Simple Mbed Cloud Client provides Greentea tests to test your Simple Mbed Cloud
...
-        "test-socket-connect"           : "net.connect(MBED_CONF_APP_WIFI_SSID, MBED_CONF_APP_WIFI_PASSWORD, NSAPI_SECURITY_WPA_WPA2);",
-        "test-block-device-object"      : "SDBlockDevice bd(D11, D12, D13, D9);"
-    }
-}
+"target.network-default-interface-type": "ETHERNET",
 ```
-For more examples and a template `mbed_app.json`, see [simple-mbed-cloud-client-template-restricted](https://github.com/ARMmbed/simple-mbed-cloud-client-template-restricted).
 
-4. You may need to delete your `main.cpp`.
+1. If your platform does not have a default storage component in `targets.json`, you can define this in `mbed_app.json`. Aditionally, you may want to double-check the storage pinout configuration for your platform in the `mbed_lib.json` file in `mbed-os/components/storage/blockdevice/`.
```
[the remark this answers is not in the document]
Comment (MarceloSalazar) [reply README.md]: IMO the person testing this shouldn't worry about storage or network configuration but instead, start with a working application.
I've removed references to storage & network config and added link to apps.
Details are shown in the porting guide in the SMCC app.

File: TESTS/host_tests/sdk_host_tests.py [hunk 4]
```diff
@@ -96,13 +97,27 @@ def setup(self):
         self.register_callback('fail_test', self._callback_fail_test)
         
         # Setup API config
-        api_key_val = os.environ.get("MBED_CLOUD_API_KEY")            
+        try:
+            result = subprocess.check_output(["mbed", "config", "-G", "MBED_CLOUD_SDK_CONFIG"], \
```
Comment (janjongboom) [inline TESTS/host_tests/sdk_host_tests.py]: What is this, and why do we have it here? This seems way too complicated.

File: TESTS/simple/connect/main.cpp [hunk 5]
```diff
@@ -39,45 +31,39 @@ void smcc_register(void) {
...
-#if defined(MBED_CONF_APP_TEST_SOCKET_OBJECT)
-    MBED_CONF_APP_TEST_SOCKET_OBJECT
-#else
-    EthernetInterface net;
-#endif
-
-#if defined(MBED_CONF_APP_TEST_SOCKET_CONNECT)
-    nsapi_error_t status = MBED_CONF_APP_TEST_SOCKET_CONNECT
+#if MBED_CONF_TARGET_NETWORK_DEFAULT_INTERFACE_TYPE == ETHERNET
+    NetworkInterface *net = NetworkInterface::get_default_instance();
```
Comment (janjongboom) [inline TESTS/simple/connect/main.cpp]: So we stopped supporting 6LoWPAN etc. now? I don't like having this in the library, I want it in the application.

File: TESTS/host_tests/sdk_host_tests.py (hunk 4 above)
Comment (MarceloSalazar) [reply TESTS/host_tests/sdk_host_tests.py]: I've just push changes to simplify this and reuse the `mbed config` option :)
Can you review?

**Comment by MarceloSalazar** [comment]
@janjongboom @bridadan if you don't mind, I'll merge this now, as are running out of time for an upcoming & critical event.
This PR brings tests and documentation enhancements, not really new features.
Feel free to send PRs or raise issues! Thanks!

**Pull request merged**